### PR TITLE
Fixes part of #1095 : Addition of Firebase SDKs 

### LINKED ISSUE #1095: Addition of Firebase SDK [GSoC'20][Milestone 1]
Setting up Firebase in the application by - 
- Registering the app on Firebase console.
- Adding the Firebase Configuration file `google-services.json` to our codebase.
- Adding basic Firebase dependencies in both project-level and app-level `build.gradle`. 
- Adding `Crashlytics` dependency and setting it up from console as well
- Enabling `Google Analytics` for our application in the Firebase console and adding its dependency in the codebase.

Important Links - 
- [Add Firebase to your Android project](https://firebase.google.com/docs/android/setup)
- [Setting up Crashlytics](https://firebase.google.com/docs/crashlytics/get-started?platform=android)
- [Setting up Analytics](https://firebase.google.com/docs/crashlytics/get-started?platform=android)

### PR DESCRIPTION
**Pull request by Sarthak2601**
## Explanation

Fixes part of #1095 
- Added classpath dependencies to the project level build.gradle 
- Added Crashlytics plugin to the app-level build.gradle
- Added dependencies to app and domain level build.gradle

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Review by vinitamurthi — APPROVED** [review]
LGTM, @rt4914 checking with you if its okay to push this PR in and have a follow up PR use these dependencies

**Comment by rt4914** [comment]
> LGTM, @rt4914 checking with you if its okay to push this PR in and have a follow up PR use these dependencies

@veena14cs @Sarthak2601 Yeah, its fine. Thanks.

File: domain/build.gradle [hunk 1]
```diff
@@ -49,7 +49,9 @@ dependencies {
       'androidx.lifecycle:lifecycle-livedata-ktx:2.2.0-alpha03',
       'com.google.dagger:dagger:2.24',
       "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version",
-      'androidx.exifinterface:exifinterface:1.0.0-rc01'
+      'androidx.exifinterface:exifinterface:1.0.0-rc01',
```
Comment (rt4914) [inline domain/build.gradle]: Nit: We are actually following alphabetical order here.

File: app/build.gradle [hunk 2]
```diff
@@ -89,6 +90,8 @@ dependencies {
       'org.jetbrains.kotlinx:kotlinx-coroutines-core:1.2.1',
       'org.jetbrains.kotlinx:kotlinx-coroutines-android:1.2.1',
       'org.mockito:mockito-core:2.7.22',
+      'com.google.firebase:firebase-analytics-ktx:17.4.2',
```
Comment (rt4914) [inline app/build.gradle]: Nit: We are actually following alphabetical order here.

**Review by rt4914 — APPROVED** [review]
LGTM. One nit change, we are following alphabetical order in dependencies.

File: app/build.gradle (hunk 2 above)
Comment (Sarthak2601) [reply app/build.gradle]: Done

File: domain/build.gradle (hunk 1 above)
Comment (Sarthak2601) [reply domain/build.gradle]: Done.

File: domain/build.gradle [hunk 3]
```diff
@@ -48,6 +48,8 @@ dependencies {
       'androidx.appcompat:appcompat:1.0.2',
       'androidx.lifecycle:lifecycle-livedata-ktx:2.2.0-alpha03',
       'com.google.dagger:dagger:2.24',
+      'com.google.firebase:firebase-analytics-ktx:17.4.2',
+      'com.google.firebase:firebase-crashlytics:17.0.0',
       "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version",
       'androidx.exifinterface:exifinterface:1.0.0-rc01'
```
Comment (rt4914) [inline domain/build.gradle]: While you are at this PR, please fix this also `'androidx.exifinterface:exifinterface:1.0.0-rc01'`

**Review by rt4914 — APPROVED** [review]
One nit change suggested. Thanks.

File: domain/build.gradle (hunk 3 above)
Comment (Sarthak2601) [reply domain/build.gradle]: Done.

**Review by rt4914 — APPROVED** [review]
LGTM.

**Pull request merged**